mc/Voyage-Mongo-Core.package/VOMongoRepositoryResolver.class/instance/basicSave.id..st

```diff
@@ -1,7 +1,13 @@
 private persistence
 basicSave: anObject id: idObject
 	| futures |
-	
+	"if enabled by repository, ensure the object to update is current."
+	repository isOnlyUpdateIfCurrent ifTrue: [
+		self 
+			basicIsCurrent: anObject
+			id: idObject
+			ifFalse: [ VOMongoSaveConflictError signal ] ].
+
```

Comment:
Sorry, that is not going to work in a reliable way. It is the classic time of check, time of action race condition.

You check.. ->
              <-  other image inserts
You compare your versions
You update ->

and you have a lost update...

https://github.com/zecke/osmo-smsc/blob/master/mc/ShortMessageCenter-Database.package/ShortMessageDatabase.class/instance/lockDestination.for..st for an example of findAndModify but db.collection.update is atomic as well.

So the update that is executed should be for the _id of the object and the version number. If not update was made, we have a conflict.